### PR TITLE
Maplibre native 6.4

### LINKED ISSUE #228: Metal support on iOS
I saw the awesome news that the [latest MapLibre iOS](https://github.com/maplibre/maplibre-gl-native-distribution/releases/tag/6.0.0) supports Metal for rendering. Is an upgrade of the iOS dependency to 6.0.0 planned soon? Really excited to see the improved rendering performance!

### PR DESCRIPTION
**Pull request by sjchristi**
## Description

Fixes #228

I'm assuming y'all will want some additional changes/fixes -- happy to integrate whatever feedback you might have.  I have tested this on simulator and on device and it all seems to work ok.  I've also tested it on my own application and that has gone well.

I haven't bumped package version or put any notes in the changelog; would appreciate some feedback on what you'd like to be done there.

I could never get husky to work so I had to remove it, but I can add it back (I kept getting errors about the husky command not being found..  I'm not sure how to fix those -- spent an hour chasing potential fixes but never found anything that worked).

## Checklist

- [X] I have tested this on a device/simulator for each compatible OS
- [X] I formatted JS and TS files with running `yarn lint:fix` in the root folder
- [X] I have run tests via `yarn test` in the root folder
- [X] I updated the documentation with running `yarn generate` in the root folder
- [X] I mentioned this change in `CHANGELOG.md`
- [X] I updated the typings files (`index.d.ts`)
- [X] I added/updated a sample (`/example`)

### CODE REVIEW

**Comment by tyrauber** [comment]
hey @sjchristi, this is fantastic!  Can you checkout and pull main, and then rebase this branch to resolve the conflicts?  Please bump to 10.0.0.alpha.4 and add a line to the top of the changelog referencing the new version and this PR.   When I get a moment, I'll checkout the branch run my own tests and then merge.

What version of React Native have you tested against. I know[ a few issues have cropped up](https://github.com/maplibre/maplibre-react-native/issues/231#issuecomment-1914849122) of people wanting to use older versions of react-native. I am wondering if we should test against previous versions as well?

**Comment by sjchristi** [comment]
> hey @sjchristi, this is fantastic! Can you checkout and pull main, and then rebase this branch to resolve the conflicts? Please bump to 10.0.0.alpha.4 and add a line to the top of the changelog referencing the new version and this PR. When I get a moment, I'll checkout the branch run my own tests and then merge.

Should be done -- please let me know if I can make any additional changes/tweaks.

> What version of React Native have you tested against. I know[ a few issues have cropped up](https://github.com/maplibre/maplibre-react-native/issues/231#issuecomment-1914849122) of people wanting to use older versions of react-native. I am wondering if we should test against previous versions as well?

I tested against react native 0.72 -- both in the sample app included here in the repo and in my own personal project.  I'm admittedly and idiot with a lot of this react native stuff (sorry) but I could try to downgrade the sample app and see if it works with earlier versions of react native.  Are there specifics versions with which we should test compatibility against?

**Comment by tyrauber** [comment]
@sjchristi, What do you think about updating all these package declarations and imports to say `com.maplibre` instead of `com.mapbox`?  Do you think we could do a massive find and replace?

File: example/ios/RNMaplibreExample.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
       "state" : {
-        "revision" : "ffda61e298c1490d4860d5184e80d618aaadc089",
-        "version" : "5.13.0"
+        "revision" : "6d0071977ed1f2380c739715f82ac650f99b0824",
+        "version" : "6.4.0"
       }
     }
   ],
-  "version" : 2
+  "version" : 3
```
Comment (tyrauber) [inline example/ios/RNMaplibreExample.xcworkspace/xcshareddata/swiftpm/Package.resolved]: @sjchristi, Not sure why, but this change here causes xcode build errors for me on Xcode 15.2. Switching this back to version 2, seems to resolve the issue.

Comment (sjchristi) [reply example/ios/RNMaplibreExample.xcworkspace/xcshareddata/swiftpm/Package.resolved]: Ok.  I'm on xcode 15.3, it probably made that switch.  I can change it back

Comment (tyrauber) [reply example/ios/RNMaplibreExample.xcworkspace/xcshareddata/swiftpm/Package.resolved]: Are you building with Xcode or the command line tools?  We need to make sure the example builds with `react-native run-ios` and `react-native run-android`.

Comment (sjchristi) [reply example/ios/RNMaplibreExample.xcworkspace/xcshareddata/swiftpm/Package.resolved]: I tested the sample app from command line.  For my person app, I tested from both command line and directly from xcode.

**Comment by sjchristi** [comment]
> @sjchristi, What do you think about updating all these package declarations and imports to say `com.maplibre` instead of `com.mapbox`? Do you think we could do a massive find and replace?

Yes, could call, we should fix this now that we're moving everything around anyhow.  Will get this integrated.

**Comment by sjchristi** [comment]
Ok, com.mapbox to com.maplibre rename is done.  Tested on the sample app and it seems good.

**Comment by sjchristi** [comment]
ok, rename from com.mapbox to com.maplibre is done.  Tested in the example app and it seemed to be good.  I'm integrating into my own personal app and will do more testing there.

**Comment by tyrauber** [comment]
@sjchristi, this is really great! Really nice work, seriously. There is a lot here, I have waded through the commits. Everything appears to be pretty staight-forward rename. I build successfully locally. Merging.

**Pull request merged**